### PR TITLE
extensions: make subcmds check/wait for respective extensions

### PR DESCRIPTION
**Pull request by Pothulapati**
This commit updates the extension subcmds to check and wait
for the respective extensions to be up before running them.

The same healthcheck pkg and respective extension checks
 are used to at the check/wait logic.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

**Comment by Pothulapati** [comment]
Depends on #5564

**Review by adleong — COMMENTED** [review]
I wonder if you could avoid adding this check to each subcommand individually by adding it to a PersistentPreRun on the `linkerd viz` parent command.  Just an idea, TIOLI.

**Comment by Pothulapati** [comment]
> I wonder if you could avoid adding this check to each subcommand individually by adding it to a PersistentPreRun on the linkerd viz parent command. Just an idea, TIOLI.

That would actually simplify the logic but I did not opt into that route for the following reasons:
- Some cmds i.e `install`, `uninstall`, etc do not actually need this functionality. So, Enforcing them with if conditions could be complex to limit it on some commands.
- Some cmds like `dashboard`, etc configure the passed timeouts passed through `--wait` into this logic too which could complex to do if the logic is made global.

WDYT?

**Review by kleimkuhler — COMMENTED** [review]
So shouldn't `jaeger/cmd.configureAndRunChecks` now also just called `checkForJaeger`? After it validates `options`, they seem to do the same thing (after #5564) merges.

**Review by adleong — APPROVED** [review]
@Pothulapati yup, that totally makes sense

File: jaeger/cmd/check.go
Comment (Pothulapati) [reply jaeger/cmd/check.go:42]: Yep, That is the plan

**Review by kleimkuhler — APPROVED** [review]

**Comment by kleimkuhler** [comment]
Approved, but again for after #5564 merges!

**Comment by Pothulapati** [comment]
@kleimkuhler Can you take one last look before I merge this?

**Review by kleimkuhler — APPROVED** [review]
🤙

**Comment by kleimkuhler** [comment]
One sec I may have a comment on this

**Comment by kleimkuhler** [comment]
Talked off GitHub, but some of these commands the checks are performed twice in a row, once by `checkForViz` and then `api.CheckClientOrExit`.

**Comment by Pothulapati** [comment]
@kleimkuhler Can you PTAL?

**Review by kleimkuhler — APPROVED** [review]
This last change looks great 👍

**Pull request merged**
